kymatio/scattering1d/backend/torch_skcuda_backend.py

```diff
@@ -149,13 +147,10 @@ def __call__(self, x, k):
         if not x.is_cuda and self.backend == 'skcuda':
             raise TypeError('Use the torch backend (without skcuda) for CPU tensors.')
 
-        if not _is_complex(x):
-            raise TypeError('The input and outputs should be complex.')
-
-        if not x.is_contiguous():
-            raise RuntimeError('Input should be contiguous.')
+        contiguous_check(x) 
+        complex_check(x)
 
-        out = x.new(x.shape[0], x.shape[1], x.shape[2] // k, 2)
+        out = torch.zeros((x.shape[0], x.shape[1], x.shape[2] // k, 2), dtype=x.dtype, layout=x.layout, device=x.device)
```

Comment:
`torch.empty`

Reply:
Can we have a discussion about the shape of `x`. Does it not have a 2 at the end? If it does have a 2 at the end, then we should have `out = torch.empty_like(x[:, :, :, ::k])`

Reply:
It does have a 2 at the end. `out = torch.empty_like(x[:, :, ::k])` works.